### DISCUSSION TITLE
How do we access indicator values after a backtest?

### DISCUSSION THREAD

**Discussion by crazy25000** [nautechsystems/nautilus_trader#335] [Q&A] [answered]
Hello, 

I looked through the docs/API reference and couldn't find anything that indicated the values are returned. For now, I've been saving the values to a list and returning it with the strategy. Since the engine already has those values, I assumed they'd be accessible.

Thanks!

Example I've been working with: https://github.com/nautechsystems/nautilus_trader/blob/master/examples/backtest/fx_ema_cross_audusd_ticks.py

**Comment by cjdsellers** [comment, marked as answer]
Hi there,

The platform makes no assumptions as to how an indicator may be used. However to support indicator based trading the platform provides: 
- An `Indicator` base class along with reference implementations for common indicators.
- Some convenience methods for registering an indicator with a `TradingStrategy` to receive various market data types.
```
register_indicator_for_quote_ticks(self, InstrumentId instrument_id, Indicator indicator)
register_indicator_for_trade_ticks(self, InstrumentId instrument_id, Indicator indicator)
register_indicator_for_bars(self, BarType bar_type, Indicator indicator)
```

Beyond this, the usage of the indicator is left to the individual user. Some users may construct a feature set and pass these to a model, others may make decisions based directly on the value, or use it as an input to an algo, some may log the output vale each time the indicator calculates, and indeed some may also choose to store all of the outputs (etc etc).

So your current approach is roughly one of the intended use cases.

So that I could assist further, what was the intended use for the indicator values?

**Reply by crazy25000** [reply]
Understood, that makes sense. For plotting - whether to debug or visually confirm logic is working as expected.
